ref/haskell/src/Codec/Binary/Bech32.hs

```diff
@@ -97,30 +97,32 @@ bech32Decode bech32 = do
     guard $ bech32VerifyChecksum hrp' dat'
     return (hrp', take (BS.length dat - 6) dat')
 
-toBase32 :: [Word8] -> [Word5]
-toBase32 dat = pad ++ result
+convertBits :: [Word] -> Int -> Int -> Bool -> Maybe [Word]
```

Comment:
It indeed felt unsettling to use the partial fromJust, and using Identity to get rid of it in fact crossed my mind, but I abstained because I felt it might be overkill (fromJust being safe here was easy to see/prove after all).

I think I'll adopt your suggestion anyway, it is hard to resist type safety :) 

About boolean blindness: I also considered just doing `data Padding = Pad | NoPad`. Here I actually don't know why I didn't do it in the end :see_no_evil: